packages/config-loader/src/sources/types.ts

```diff
@@ -89,3 +90,14 @@ export interface ConfigSource {
  * @public
  */
 export type SubstitutionFunc = (name: string) => Promise<string | undefined>;
+
+/**
+ * A custom function to be used for parsing configuration content.
+ *
+ * @remarks
+ *
+ * The default parsing function will parse configuration content as yaml.
+ *
+ * @public
+ */
+export type ParsingFunc = (contents: string) => JsonObject | undefined;
```

Comment:
question: should this be asynchronous?

Reply:
Probably right? 

I think it also probably makes sense to wrap this up a little bit. Let's make the type look something more like this:

```ts
export type Parser = ({ contents }: { contents: string }) => Promise<{ result?: JsonObject }>
```

Thinking that it let's us evolve the API a little bit over time if needed without breaking it.

Reply:
Updated to new API